src/vcpkg/commands.build.cpp

```diff
@@ -1488,54 +1488,75 @@ namespace vcpkg
         return res;
     }
 
-    struct CreateLogDetails
+    static void append_log(const Filesystem& fs, const Path& path, int max_size, std::string& out)
     {
-        const Filesystem& fs;
+        max_size -= static_cast<int>(path.native().size()) + 80 /* size of all static text */;
```

Comment:
Instead of a comment can this be done statically (e.g. by forming the static text into `StringLiteral`s and adding their `size()`s together)?

What about the number of skipped lines, which won't be a fixed length?

Reply:
> What about the number of skipped lines, which won't be a fixed length?

I have added 5 Characters for that. But I will make this code more obvious.